core/appeal/service.go

```diff
@@ -218,11 +218,19 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
 			return fmt.Errorf("retrieving provider: %w", err)
 		}
 
-		if err := s.checkExtensionEligibility(appeal, provider, activeGrants); err != nil {
+		var err2 error
+		var policy *domain.Policy
+		if isAdditionalAppealCreation && appeal.PolicyID != "" && appeal.PolicyVersion != 0 {
+			policy = policies[appeal.PolicyID][appeal.PolicyVersion]
+		} else {
+			policy, err2 = getPolicy(appeal, provider, policies)
```

Comment:
why not return error here instead if `err2 != nil`

Reply:
yea, initially i did that, but unfortunately breaking existing tests due to validation order,
so i keep the existing behavior by keeping validation order the same,
do you have any suggestion mas @mabdh ?

Reply:
Hmm that is fair enough. But seeing the flow, the validation is unnecessary if `policy` is nil right ? 
Should we also re-evaluate the testcase itself?